core/indigo-core/reaction/src/reaction_multistep_detector.cpp

```diff
@@ -236,15 +237,54 @@ void ReactionMultistepDetector::createSummBlocks()
     }
 }
 
+void ReactionMultistepDetector::sortSummblocks()
+{
+    // Create a list of original indices
+    std::vector<int> indices(_component_summ_blocks.size());
+    std::iota(indices.begin(), indices.end(), 0);
+
+    // Sort the indices based on reaction_idx
+    std::sort(indices.begin(), indices.end(),
+              [&](int a, int b) -> bool { return _component_summ_blocks[a].reaction_idx < _component_summ_blocks[b].reaction_idx; });
+
+    // Create a mapping from old index to new index
+    std::vector<int> old_to_new(_component_summ_blocks.size());
+    for (int new_idx = 0; new_idx < static_cast<int>(indices.size()); ++new_idx)
+        old_to_new[indices[new_idx]] = new_idx;
+
+    // Sort the blocks based on the sorted indices
+    std::vector<MolSumm> sorted_blocks;
+    sorted_blocks.reserve(_component_summ_blocks.size());
+    for (auto idx : indices)
+        sorted_blocks.emplace_back(std::move(_component_summ_blocks[idx]));
+    _component_summ_blocks = std::move(sorted_blocks);
+
+    // Update arrows_to and arrows_from with new indices
+    for (auto& block : _component_summ_blocks)
```

Comment:
seems to be error. Can't update loop element

Reply:
updated